src/map/lua/lua_baseentity.cpp

```diff
@@ -8625,6 +8625,41 @@ inline int32 CLuaBaseEntity::getAllianceSize(lua_State* L)
     return 1;
 }
 
+/************************************************************************
+*  Function: getAllianceID()
+*  Purpose : Returns the id of the alliance/party or player
+*  Example : local allianceId = player:getAllianceID()
+*  Notes   :
+************************************************************************/
+
+inline int32 CLuaBaseEntity::getAllianceID(lua_State* L)
+{
+    DSP_DEBUG_BREAK_IF(m_PBaseEntity == nullptr);
+
+    uint8 id = 0;
+
+    auto PChar = dynamic_cast<CCharEntity*>(m_PBaseEntity);
+
+    if (PChar != nullptr)
+    {
+        if (PChar->PParty && PChar->PParty->m_PAlliance)
+        {
+            id = PChar->PParty->m_PAlliance->m_AllianceID;
+        }
+        else if (PChar->PParty)
```

Comment:
Potentially doing the same conditional twice in here.

You can wrap all of this in a PParty check, then check if the character is in an alliance.

If not, you should really be pushing nil. Falling back to party id and then char id is not intuitive behavior.